uni/lib/controller/parsers/parser_schedule_html.dart

```diff
@@ -64,6 +70,45 @@ Future<List<Lecture>> getScheduleFromHtml(http.Response response) async {
       semana = semana.expand((i) => [(i - 1) < 0 ? 0 : i - 1]).toList();
     }
   });
+  final overlappingClasses = document.querySelectorAll('.dados > tbody > .d');
```

Comment:
Also, try seperating the added feature to a private function, this functions is huge